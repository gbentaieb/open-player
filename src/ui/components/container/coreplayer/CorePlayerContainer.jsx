import { connect } from 'react-redux';
import RxPlayer from 'rx-player';
import CorePlayer from '../../presentational/coreplayer/CorePlayer';
import { setPlayerState } from '../../../actions/CoreActions';

function mapStateToProps(state) {
  return {
    RxPlayer,
    url: state.config.url,
    playRequested: state.core.playRequested,
  };
}

export default connect(
  mapStateToProps,
  {
    setPlayerState,
  },
)(CorePlayer);
