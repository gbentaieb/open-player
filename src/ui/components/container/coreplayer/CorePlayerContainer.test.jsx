import React from 'react';
import { createMockStore } from 'redux-test-utils';

import FakeRxPlayer from '../../../../../test/fakeRxPlayer';
import CorePlayerContainer from './CorePlayerContainer';

const defaultStore = {
  config: {
    url: 'https://www.test.com/test.mp4',
  },
  core: {
    playRequested: true,
  },
};

const defaultProps = {
  RxPlayer: FakeRxPlayer,
};

describe('CorePlayerContainer snapshot', () => {
  test('CorePlayerContainer > Snapshot', () => {
    const store = createMockStore(defaultStore);
    const wrapper = shallow(<CorePlayerContainer store={store} {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
