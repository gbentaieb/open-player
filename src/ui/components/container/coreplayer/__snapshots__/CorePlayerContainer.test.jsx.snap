// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CorePlayerContainer snapshot CorePlayerContainer > Snapshot 1`] = `
<CorePlayer
  RxPlayer={[Function]}
  playRequested={true}
  setPlayerState={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getAction": [Function],
      "getActions": [Function],
      "getState": [Function],
      "isActionDispatched": [Function],
      "isActionTypeDispatched": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getAction": [Function],
        "getActions": [Function],
        "getState": [Function],
        "isActionDispatched": [Function],
        "isActionTypeDispatched": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": undefined,
    }
  }
  url="https://www.test.com/test.mp4"
/>
`;
